drivers/bbram/bbram_it8xxx2.c

```diff
@@ -53,7 +59,30 @@ static const struct bbram_driver_api bbram_it8xxx2_driver_api = {
 
 static int bbram_it8xxx2_init(const struct device *dev)
 {
-	ARG_UNUSED(dev);
+	const struct bbram_it8xxx2_config *config = dev->config;
+	volatile uint8_t *base_addr = (uint8_t *)config->base_addr;
+	volatile uint8_t *bram_valid_flag0 = base_addr + BRAM_IDX_VALID_FLAGS0;
+	volatile uint8_t *bram_valid_flag1 = base_addr + BRAM_IDX_VALID_FLAGS1;
+	volatile uint8_t *bram_valid_flag2 = base_addr + BRAM_IDX_VALID_FLAGS2;
+	volatile uint8_t *bram_valid_flag3 = base_addr + BRAM_IDX_VALID_FLAGS3;
```

Comment:
Why is volatile needed for these pointers?
    config->base_addr should not change and therefore removing optimization on base_addr for no reason.
    In this code, there is no redundant accesses to optimize out multiple reads that needs to be managed.

Reply:
Done